azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/options/AzureComputeArmTemplateOptions.java

```diff
@@ -0,0 +1,419 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.compute.options;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import java.util.List;
+import java.util.Map;
+
+import org.jclouds.compute.options.TemplateOptions;
+import org.jclouds.domain.LoginCredentials;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.scriptbuilder.domain.Statement;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Contains options supported by the {@link org.jclouds.compute.ComputeService#createNodesInGroup(
+ * String, int, org.jclouds.compute.options.TemplateOptions)} operation.
+ *
+ * <h2>Usage</h2> The recommended way to instantiate a {@link AzureComputeArmTemplateOptions} object is to statically
+ * import {@code AzureComputeArmTemplateOptions.*} and invoke a static creation method followed by an instance mutator (if
+ * needed):
+ * <p>
+ *
+ * <pre>
+ * import static org.jclouds.compute.options.AzureComputeArmTemplateOptions.Builder.*;
+ * ComputeService client = // get connection
+ * templateBuilder.options(inboundPorts(22, 80, 8080, 443));
+ * Set&lt;? extends NodeMetadata&gt; set = client.createNodesInGroup(tag, 2, templateBuilder.build());
+ * </pre>
+ *
+ */
+public class AzureComputeArmTemplateOptions extends TemplateOptions implements Cloneable {
+
+   protected String virtualNetworkName;
+   protected List<String> subnetNames = ImmutableList.of();
```

Comment:
The template options already have a way to provide the IDs of the networks where the VMs have to be attached. Can that property be used instead of having these ones?